File: server/Models/Domain/Project.cs
using System;
using server.Models.Domain.Enums;

namespace server.Models.Domain;

public record class Project
{
    public int ProjectId { get; init; }
    public string Name { get; init; }
    public string Description { get; init; } = string.Empty;
    public ProjectType ProjectType { get; init; }
    public ProjectStatus Status { get; init; }

    public string? OwnerId { get; init; }
    public string? AnnotationGuidelinesUrl { get; init; }

    public DateTime CreatedAt { get; init; }
    public DateTime UpdatedAt { get; init; }

    public virtual ApplicationUser? Owner { get; init; }

    public virtual ICollection<LabelScheme> LabelSchemes { get; init; } = [];
    public virtual ICollection<DataSource> DataSources { get; init; } = [];
    public virtual ICollection<Asset> Assets { get; init; } = [];
    public virtual ICollection<Workflow> Workflows { get; init; } = [];
    public virtual ICollection<ProjectMember> ProjectMembers { get; init; } = [];
}

File: frontend/src/types/project/project.ts
// Domain enums - should match backend exactly
export enum ProjectType {
    OTHER = 'OTHER',
    IMAGE_CLASSIFICATION = 'IMAGE_CLASSIFICATION',
    OBJECT_DETECTION = 'OBJECT_DETECTION',
    IMAGE_SEGMENTATION = 'IMAGE_SEGMENTATION',
    VIDEO_ANNOTATION = 'VIDEO_ANNOTATION',
    TEXT_ANNOTATION = 'TEXT_ANNOTATION',
}

export enum ProjectStatus {
    ACTIVE = 'ACTIVE',
    ARCHIVED = 'ARCHIVED',
    READ_ONLY = 'READ_ONLY',
    PENDING_DELETION = 'PENDING_DELETION',
}

// Main Project interface
export interface Project {
    id: number;
    name: string;
    description: string;
    projectType: ProjectType;
    status: ProjectStatus;
    createdAt: string;
    updatedAt: string;
    ownerId?: string;
    annotationGuidelinesUrl?: string;
}
